prototype2/gdgem/vmm3/CalibrationFile.cpp

```diff
@@ -0,0 +1,89 @@
+/* Copyright (C) 2018 European Spallation Source, ERIC. See LICENSE file */
+//===----------------------------------------------------------------------===//
+///
+/// \file
+///
+/// \brief Class for handling calibration files for VMM asics
+///
+//===----------------------------------------------------------------------===//
+
+#include <common/Log.h>
+#include <common/Trace.h>
+#include <gdgem/vmm3/CalibrationFile.h>
+#include <nlohmann/json.hpp>
+#include <fstream>
+
+using json = nlohmann::json;
+
+// #undef TRC_LEVEL
+// #define TRC_LEVEL TRC_L_DEB
+
+
+/// \brief clear the calibration array
+CalibrationFile::CalibrationFile() {
+  auto n = sizeof(calibrations)/sizeof(calibration);
+  for (size_t i = 0; i < n; i++) {
+    ((calibration *)calibrations)[i] = nocorr;
+  }
+}
+
+
+/// \brief load calibration from file
+CalibrationFile::CalibrationFile(std::string jsonfile) : CalibrationFile() {
+  std::ifstream t(jsonfile);
+  std::string jsonstring((std::istreambuf_iterator<char>(t)),
+                  std::istreambuf_iterator<char>());
+
+  loadCalibration(jsonstring);
+}
+
+
+/// \brief parse json string with calibration data
+void CalibrationFile::loadCalibration(std::string jsonstring) {
+  nlohmann::json root;
+  try {
+    root = nlohmann::json::parse(jsonstring);
+  }
+  catch (...) {
+    LOG(Sev::Warning, "Invalid Json file: {}", jsonstring);
+    return;
+  }
+
+  auto vmmcal = root["vmm_calibration"];
+  for (unsigned int i = 0; i < vmmcal.size(); i++) {
```

Comment:
It looks like a range based for loop (`for (auto &Item : vmmcal)`) would be excellent here.

Reply:
ok, can try that.